docs/providers.rst

```diff
@@ -20,34 +20,34 @@ Creating providers
 ==================
 
 Providers are responsible for building dependencies needed by Injectable
-objects. A good example would be the ``ResponseDataProvider``,
-which builds and provides a ``ResponseData`` instance for Injectables
-that need it, like the ``ItemWebPage``.
+objects. A good example would be the ``HttpResponseProvider``,
+which builds and provides a ``web_poet.HttpResponse`` instance for Injectables
+that need it, like the ``web_poet.ItemWebPage``.
 
 .. code-block:: python
 
     import attr
     from typing import Set, Callable
 
+    import web_poet
     from scrapy_poet.page_input_providers import PageObjectInputProvider
     from scrapy import Response
 
 
-    @attr.define
-    class ResponseData:
-        """Represents a response containing its URL and HTML content."""
-        url: str
-        html: str
-
-
-    class ResponseDataProvider(PageObjectInputProvider):
-        """This class provides ``web_poet.page_inputs.ResponseData`` instances."""
-        provided_classes = {ResponseData}
+    class HttpResponseProvider(PageObjectInputProvider):
+        """This class provides ``web_poet.HttpResponse`` instances."""
```

Comment:
It could be good to also have a way to provide HttpResponseBody, HttpResponseHeaders, and similar classes.

We may even go a step further, and add HttpResponseUrl and HttpResponseStatus classes; in this case, a provider for HttpResponse won't be needed at all, as andi would be able to construct it from other providers.
We don't have to do it now :)

Reply:
Hi @kmike , to clarify, do you mean adding providers for `HttpResponseBody` and `HttpResponseHeaders` inside the docs as an example? or actually adding such providers in **providers.py**?

For the latter, I'm not quite sure what would be the use case of having individual providers for each subcomponent of `HttpResponse` as it's already encompassed within the class. Although I might be missing something here.

Reply:
hey @BurnzZ! I was thinking about having providers in providers.py. The use case is to support POs which define HttpResponseBody dependencies, not HttpResponse. It doesn't make much sense here, but such POs may be written to use other sources of inputs (e.g. Zyte API), and in this case the distinction could be more important. For example, it's possible to save a bit of API traffic by not requesting HttpResponseHeaders if they're not needed.

Reply:
Gotcha! That makes sense! I've created a new issue in https://github.com/scrapinghub/scrapy-poet/issues/70 to track this idea down the line. :)